### ISSUE TITLE
gpg fails to use yubikey

### ISSUE THREAD

**Issue by srhb** [NixOS/nixpkgs#85127]
There appears to be a problem (possibly related to 59281f742a0ce8def71be3576129b4178ec579bb) communicating with the yubikey when using libusb-compat.

**To Reproduce**

```
❯ gpg --card-status
gpg: selecting card failed: No such device
gpg: OpenPGP card not available: No such device
```

Then, using a gpg with `libusb = libusb1`

```
❯ pkill gpg-agent
❯ $(nix-build --no-out-link -E 'with import <nixpkgs> {}; gnupg.override { libusb = libusb1; }')/bin/gpg --card-status
[ gpg finds the card and works correctly ]
```

**Comment by sylv-io** [comment]
did you enable pcscd deamon ?

In my case I also added the line `reader-port Yubico Yubi` to `.gnupg/scdaemon.conf` (See [Troubleshooting Issues with GPG](https://support.yubico.com/support/solutions/articles/15000011370-troubleshooting-issues-with-gpg)).

**Comment by srhb** [comment]
No, I'm explicitly not using pcscd, because I find it annoying in my workflow -- I haven't tested if that method is also affected. This is a regression regardless, however. :)

**Comment by srhb** [comment]
(If I was unclear in my description, the second example makes everything work again like it used to, without pcscd.)

**Comment by sylv-io** [comment]
my bad. I didn't read your description correctly.
Using libusb instead of pcsc sounds great, I will adapt my config :)

**Comment by jpotier** [comment]
Oh I just fell back on pcscd and it worked again, so I didn't look further…

**Comment by MasseR** [comment]
`pcscd` worked only for a while for me. I can't seem to get it to work at all with the current unstable channel.

**Comment by jpotier** [comment]
Same, pcscd trick sorta stopped working :(

**Comment by jpotier** [comment]
Well, this:

```sh
$(nix-build --no-out-link -E 'with import <nixpkgs> {}; gnupg.override { libusb = libusb1; }')/bin/gpg --card-status
```

now fails (without pcscd daemon)

**Comment by snglth** [comment]
It also applies to Librem Key. I guess new libusb-compat breaks just any usb smartcard.

**Comment by johnae** [comment]
Broke my setup as well. I don't use pcscd. With pcscd I could get gnupg working again (but in an annoying way because it has never worked well and is also pointless as it doesn't provide any value). The suggested fix by @srhb is working for me.

**Comment by johnae** [comment]
Latest unstable is still broken but now it seems the fix here doesn't help anymore.

**Comment by MasseR** [comment]
Fwiw, this was a blocker for me so I had to switch the channel back to stable.

**Comment by jpotier** [comment]
I'm currently working on unstable with pcscd daemon, which regularly fails when I plug-in my key, but works after being forcibly restarted :shrug: Not great, but can work with that.

**Comment by srhb** [comment]
Seems to be fixed for me with 6673a4988e4a4ff25ca94a84d8e2acf92453dbf1. I'll verify later and close this.

**Comment by jpotier** [comment]
Can also confirm that it works now on unstable. I'm on `20.09pre222973.7c399a4ee08 (Nightingale)`, which includes 6673a49. Successfully disabled pcscd daemon.
